scripts/buildArticles.js

```diff
@@ -0,0 +1,143 @@
+const fs = require("fs");
+const path = require("path");
+const matter = require("gray-matter");
+const glob = require("glob");
+
+const articlesDirectory = path.join(process.cwd(), "..", "src/pages/");
+
+fs.readFile("../src/configs/categories.json", (err, data) => {
+  if (err) throw err;
+  const categories = JSON.parse(data);
+
+  const getDirectories = (src, callback) => glob(src + "/**/*", callback);
+
+  const articles = [];
+
+  categories.forEach((element) => {
+    const articleGroupedByCategory = [];
+    const category = element["slug"];
+
+    const article = {};
+    article["category"] = element["slug"];
+    article["article"] = true;
+    article["url"] = "/" + element["slug"] + "/";
+    article["path"] = [article["url"]];
+    article["items"] = [];
+
+    if (
+      fs.existsSync(
+        articlesDirectory + article["url"].substring(1) + "index.mdx"
+      )
+    ) {
+      const fileContents = fs.readFileSync(
+        articlesDirectory + article["url"].substring(1) + "index.mdx",
+        "utf8"
+      );
+      const { data, content } = matter(fileContents);
+
+      article["title"] = data["title"];
+    } else {
+      article["title"] = element["slug"];
+    }
+
+    articleGroupedByCategory.push(article);
+
+    element.items.forEach((item) => {
+      const article = {};
```

Comment:
Can we create a function which is formatting article so you could use it here and also in line 20. I know there are some differences there, but you should be able to make some generic.